azurerm/internal/services/videoanalyzer/video_analyzer_resource.go

```diff
@@ -0,0 +1,321 @@
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"type": {
+							Type:             pluginsdk.TypeString,
+							Required:         true,
+							DiffSuppressFunc: suppress.CaseDifference,
+							ValidateFunc: validation.StringInSlice([]string{
+								string("UserAssigned"),
+							}, true),
+						},
+						"identity_ids": {
+							Type:     pluginsdk.TypeSet,
+							Required: true,
+							MinItems: 1,
+							Elem: &pluginsdk.Schema{
+								Type:         pluginsdk.TypeString,
+								ValidateFunc: validate.UserAssignedIdentityID,
+							},
+						},
+					},
+				},
+			},
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceVideoAnalyzerCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).VideoAnalyzer.VideoAnalyzersClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceId := parse.NewVideoAnalyzerID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceId.ResourceGroup, resourceId.Name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for existing %s: %+v", resourceId, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_video_analyzer", resourceId.ID())
+		}
+	}
+
+	identity, err := expandAzureRmVideoAnalyzerIdentity(d)
+	if err != nil {
+		return err
+	}
+	parameters := videoanalyzer.Model{
+		PropertiesType: &videoanalyzer.PropertiesType{
+			StorageAccounts: expandVideoAnalyzerStorageAccounts(d),
+		},
+		Location: utils.String(azure.NormalizeLocation(d.Get("location").(string))),
+		Identity: identity,
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceId.ResourceGroup, resourceId.Name, parameters); err != nil {
+		return fmt.Errorf("creating %s: %+v", resourceId, err)
+	}
+
+	d.SetId(resourceId.ID())
+	return resourceVideoAnalyzerRead(d, meta)
+}
+
+func resourceVideoAnalyzerRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).VideoAnalyzer.VideoAnalyzersClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.VideoAnalyzerID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Video Analyzer %q was not found in Resource Group %q - removing from state", id.Name, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Video Analyzer %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
```

Comment:
the ID formatter's `String()` method means we can reduce this to:

```suggestion
		return fmt.Errorf("retrieving %s: %+v", *id, err)
```